cleanup

CameronHenshaw opened the pull request:
WOOOOOOO IT WORKS
Let me know if any tweaks still need to be made. I'll take another look in the morning too to see if I see anything I've missed.

ejaxon reviewed (changes requested):
All correct now, but there are a few cleanup things. Two have to do with thinking about what level of the code should have what knowledge (3 and 4 below).

1. Delete logging statement on line 29
2. O line 45, ++j should be changed to j += 1 (ESLint complaining)
3. addInfo: I think this should not return the whole result structure (why should this routine know about the structure of the API return?), just the result.result part. So instead of creating the info variable as you do, just make line 42 be `const result = {` and delete `info.` from the subsequent references. Then line 127 becomes:
```
  result.result = await addInfo(res, fields, available);
```
4. addTags: This has the opposite problem - the main routine is expected to know the structure of the return from the Postgres call. The addTags routine should just return the array of tags (or []). So the loop in the main routine starting on line 139 should move up to addTags, except that you'd just create a local ```tags``` variable in addTags and build and return that.
5. The initialization of the result structure on line 82 should set result.result to null (that was my bad decision originally). Then you can delete lines 94 and 104 (you still need 134 unless you hold the result in a separate variable and put it in the return structure at the end).

Let me know if you want to talk through any of this.

CameronHenshaw commented:
Changes made. I also changed a let declaration to a const and removed a few lines of code at the end that were checking whether result was an error or not (In a try catch before that, I was already returning result when it was an error, so I don't think those few lines of code are needed). Looping in Jon too just in case he feels like taking a look and offering any feedback.

ejaxon reviewed (approved):
Looks good!
